js_modules/dagster-ui/packages/ui-core/src/assets/useAssetCatalogFiltering.tsx

```diff
@@ -0,0 +1,126 @@
+import {TextInput} from '@dagster-io/ui-components';
+import * as React from 'react';
+
+import {useAssetGroupSelectorsForAssets} from './AssetGroupSuggest';
+import {AssetTableFragment} from './types/AssetTableFragment.types';
+import {useAssetDefinitionFilterState} from './useAssetDefinitionFilterState';
+import {useAssetSearch} from './useAssetSearch';
+import {CloudOSSContext} from '../app/CloudOSSContext';
+import {useQueryPersistedState} from '../hooks/useQueryPersistedState';
+import {useFilters} from '../ui/Filters';
+import {useAssetGroupFilter} from '../ui/Filters/useAssetGroupFilter';
+import {useAssetOwnerFilter, useAssetOwnersForAssets} from '../ui/Filters/useAssetOwnerFilter';
+import {useAssetTagFilter, useAssetTagsForAssets} from '../ui/Filters/useAssetTagFilter';
+import {useChangedFilter} from '../ui/Filters/useChangedFilter';
+import {useCodeLocationFilter} from '../ui/Filters/useCodeLocationFilter';
+import {
+  useAssetKindTagsForAssets,
+  useComputeKindTagFilter,
+} from '../ui/Filters/useComputeKindTagFilter';
+import {FilterObject} from '../ui/Filters/useFilter';
+import {WorkspaceContext} from '../workspace/WorkspaceContext';
+
+const EMPTY_ARRAY: any[] = [];
+
+export function useAssetCatalogFiltering(
+  assets: AssetTableFragment[] | undefined,
+  prefixPath: string[],
+) {
+  const [search, setSearch] = useQueryPersistedState<string | undefined>({queryKey: 'q'});
+
+  const {
+    filters,
+    filterFn,
+    setAssetTags,
+    setChangedInBranch,
+    setComputeKindTags,
+    setGroups,
+    setOwners,
+    setRepos,
+  } = useAssetDefinitionFilterState();
+
+  const searchPath = (search || '')
+    .replace(/(( ?> ?)|\.|\/)/g, '/')
+    .toLowerCase()
+    .trim();
+
+  const pathMatches = useAssetSearch(
+    searchPath,
+    assets ?? (EMPTY_ARRAY as NonNullable<typeof assets>),
+  );
+
+  const filtered = React.useMemo(
+    () => pathMatches.filter((a) => filterFn(a.definition ?? {})),
```

Comment:
Should this also take into account the search string?

Reply:
Yeah it does one layer down in `useAssetSearch` above -- honestly probably need to flatten a few of these hooks in a follow-up!